src/tools/pub/dependencies.md

```diff
@@ -442,6 +439,32 @@ To identify dependencies in your app or package that
 aren't on the latest stable versions,
 use [`dart pub outdated`][].
 
+### Use tight dev_dependencies
+
+Dependencies can be loose, specifying dependency at the highest version digit
+like `^2.0.0`, or tight, specifying dependency at the lowest version digit like
+`^1.2.4`.
+
+In your `pubspec.yaml` file, follow these recommendations.
```

Comment:
I'm not sure we need to mention this as this whole section is about dependencies in the `pubspec.yaml` file and recommendations for those dependencies. I think we can just focus on the best practices.